### PR TITLE
Delete attainment functionality in frontend

### PR DESCRIPTION
**Pull request by viivi-uhari**
Functionality for deleting attainments from the DB via frontend. When an attainment is deleted all its sub-attainments are also deleted.

### CODE REVIEW

**Review by kiiamakir — APPROVED** [review]
Looks good! Testing was a bit difficult since the get attainments route hasn't been implemented. We should get this to main before the formula attainment refactor PR so that the refactor branch can be modified accordingly.

File: client/src/components/EditAssignmentView.js [hunk 1]
```diff
@@ -66,6 +67,21 @@ const EditAssignmentView = () => {
     }
   };
 
+  // Function to delete data from the database or from the context
+  const deleteAttainment = async (attainmentId) => {
+    // If this view is opened from the course view, delete from DB
+    // Else the attainment is being edited during the creation of an instance so only delete from the context
+    if (instanceId) {
```
Comment (marillam) [inline client/src/components/EditAssignmentView.js:74]: Should this be inside try/catch? Or is error handling done somewhere else?

**Review by marillam — CHANGES_REQUESTED** [review]
Looks good for the most part! Waiting for the get attainment api to work so this is easier to test in the UI but this is a good start

File: client/src/components/EditAssignmentView.js (hunk 1 above)
Comment (viivi-uhari) [reply client/src/components/EditAssignmentView.js:74]: Added try/catch

**Review by marillam — APPROVED** [review]

**Pull request merged**